client/app/components/Input/InputTag.jsx

```diff
@@ -65,9 +65,8 @@ export class InputTag extends React.Component<Props, State> {
     const inputValue = autocompleteLabel.trim().toLowerCase();
     const inputLength = inputValue.length;
   
```

Comment:
I had tried fixing 
`Error -------------------------------------------------------------------------- app/components/Input/InputTag.jsx:68:37

Missing type annotation for `T`. `T` is a type parameter declared in array type [1] and was implicitly instantiated at
call of method `filter` [2].
`
This current solution obviously didn't work.

I was thinking maybe if I add the following code at line 67 it might pass this error.

`    const { checkboxes } = this.state;
`

Reply:
Why exactly are you calling `.slice(0, inputLength)` on `checkbox.label.toLowerCase()`? Shouldn't using `checkbox.label.toLowerCase()` be enough?

Reply:
So the issue is that `checkbox.label.toLowerCase().slice(0, inputLength) === inputValue)` returns a value of type `Checkbox[]`.

`this.getSuggestions()` is being used to update the `this.state.suggestions`, which is expecting a value of type `string[]`.
```
onSuggestionsFetchRequested = ({ value }: {value: string}) => {
    this.setState({
      suggestions: this.getSuggestions(value)
    });
  };
```

Reply:
Yep you're definitely right about changing this.state.suggestions to type Checkbox[]. That solves the other errors. But the following error remains:

Missing type annotation for T. T is a type parameter declared in array type [1] and was implicitly instantiated at
call of method filter [2].

I am reading some through other issues that people have had. Hopefully I'll run into a suitable solution soon :)

Reply:
@julianguyen this error has been resolved!